### PR TITLE
Add group rule endpoints

### PR DESCRIPTION
**Pull request by hectron**
## Problem

The Okta API supports [group rules](https://developer.okta.com/docs/reference/api/groups/#group-rule-operations), but the client is missing them.
 
## Proposal

- Add the group rule endpoints

###  Other fixes

The spec helper didn't have a way to mock which server to communicate to. I don't have access to `okta-test.okta.com`, so I used environment variables to talk to my instance of Okta.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -41,3 +45,11 @@ module TestClient
 def test_okta_token
   ENV.fetch('OKTA_TEST_TOKEN', 'x' * 40)
 end
+
+def test_okta_organization
+  ENV.fetch('OKTA_TEST_ORGANIZATION', 'okta-test')
+end
+
+def test_okta_api_endpoint
```
Comment (hectron) [inline spec/spec_helper.rb]: If this is not set, it is ok. The client handles building the API endpoint using the organization above.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -31,7 +31,11 @@ def serializable_body(*)
 
 module TestClient
   extend RSpec::SharedContext
-  let(:client) { Oktakit::Client.new(token: test_okta_token, organization: 'okta-test') }
+  let(:client) do
+    Oktakit::Client.new(token:        test_okta_token,
+                        organization: test_okta_organization,
+                        api_endpoint: test_okta_api_endpoint)
+  end
```
Comment (hectron) [inline spec/spec_helper.rb]: I didn't spend too much time trying to figure out how to scrub the okta API URL, which would make creating VCRs much easier.

**Comment by hectron** [comment]
It seems like the builds are failing due to Rubocop rules. If I have some time, I'll try to open a PR to address those and base this PR off of that fix.

**Comment by PeterCat12** [comment]
+1

**Comment by RootTJNII** [comment]
> It seems like the builds are failing due to Rubocop rules. If I have some time, I'll try to open a PR to address those and base this PR off of that fix.

https://github.com/Shopify/oktakit/pull/23

**Comment by jonpulsifer** [comment]
Hey @hectron, thanks for the PR. Unfortunately it's very big and includes more changes than you described in the title.

Can you:

1. Remove all changes that aren't related to `Add group rule endpoints` (break this PR into multiple ones that we may consider them)
1. Rebase off the latest master branch

**Pull request closed without merging**